### PR TITLE
Inital commit of Get-Dba Uptime

### PR DESCRIPTION
**Pull request by Stuart-Moore**
Fixes # 

Changes proposed in this pull request:
Add new function Get-DbaUptim

How to test this code: 
- [ ] 
- [ ] 

Has been tested on minimum requirements:
- [ ]  Powershell 3
- [ ]  Windows 7
- [ ]  SQL Server 2000

Has been tested on maximum requirements:
- [ ]  SQL Server vNext
- [ ]  Windows 10
- [ ]  Azure Database

Tests for tester:
- [ ] Working/useful help content, including link to command on dbatools web site
- [ ] All examples work as advertised
- [ ] Does not contain template content
- [ ] Does not contain excessive/unnecessary amounts of comments
- [ ] Works remotely
- [ ] Works locally
- [ ] Works on lower versions or throws error specifying version not supported
- [ ] Works with named instances
- [ ] Works with clustered instances
- [ ] Handles offline/read only databases
- [ ] Supports multiple servers (at the command line or piped from Get-SqlRegisteredServerName)
- [ ] No un-handled errors which stop the command working with multiple servers

### CODE REVIEW

File: functions/Get-DBAUptime.ps1 [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+		foreach ($servername in $SqlServer)
+		{
+            write-verbose "$functionname - server = $servername"
+			try
+			{
+				$server = Connect-SqlServer -SqlServer "$servername" -SqlCredential $Credential
+			}
+			catch
+			{
+				if ($servercount -eq 1)
```
Comment (PowerDBAKlaas) [inline functions/Get-DBAUptime.ps1]: This is the old way to catch a failed connection.
We changed to:
```
		Write-Verbose "Connecting to $SqlServer"
		try
		{
			$server = Connect-SqlServer -SqlServer $SqlServer -SqlCredential $Credential -ErrorVariable ConnectError
			
		}
		catch
		{
			Write-Warning $_
			continue
		}
```
in newer functions

File: functions/Get-DBAUptime.ps1 [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+					catch [System.Exception] {
+						Write-Exception $_
+						#Skip the windows results as they'll either be garbage or not there.
+						$SQLOnly = $true
+					}
+
+				}
+				if ($SQLOnly -eq $true)
+				{
+					$null = $collection.Add([PSCustomObject]@{
```
Comment (PowerDBAKlaas) [inline functions/Get-DBAUptime.ps1]: Put an Object that doesn't need alterations in the pipeline immediately.
No need to build a collection.

File: functions/Get-DBAUptime.ps1 [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	)
+
+    	BEGIN
+	{
+        $functionname = "Get-DBAUptime"
+		$collection = New-Object System.Collections.ArrayList
+	}
+    	PROCESS
+	{
+		$servercount = ++$i
```
Comment (PowerDBAKlaas) [inline functions/Get-DBAUptime.ps1]: There's no need for a servercount, we can use non-terminating statements even when there's only one servername.

File: functions/Get-DBAUptime.ps1 [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+					if ($ClusterCheck -eq 'Not a clustered instance' )
+					{
+						$WindowsServerName = ($servername.split("\"))[0]
+					}
+					else
+					{
+						$WindowsServerName = $ClusterCheck
+					}
+					try {
+						$WinBootTime = (Get-CimInstance -ClassName win32_operatingsystem -ComputerName $windowsServerName).lastbootuptime
```
Comment (PowerDBAKlaas) [inline functions/Get-DBAUptime.ps1]: when Get-Ciminstance fails, a fall back method with Cim-Session and -protocol DCom is plan B.

File: functions/Get-DBAUptime.ps1 [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+				if ($SQLOnly -eq $true)
+				{
+					$null = $collection.Add([PSCustomObject]@{
+							SQLServer = $servername
+							SQLStartTime = $SQLStartTime
+							SQLUptimeString = $SQLUptimeString
+							SQLUptime = $SQLUptime
+					})
+				}else{
+					$null = $collection.Add([PSCustomObject]@{
```
Comment (PowerDBAKlaas) [inline functions/Get-DBAUptime.ps1]: We'll use ComputerName = $server.NetName, InstanceName = $server.ServiceName and SqlInstance = $server.Name as output in all functions that gather information at Sql Instance level.

File: functions/Get-DBAUptime.ps1 [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+							WindowsUptime = $WindowsUptime
+							WindowsUptimeString = $WindowsUptimeString
+					})
+				}
+
+        }
+    }
+    END 
+    {
+            return $collection
```
Comment (PowerDBAKlaas) [inline functions/Get-DBAUptime.ps1]: no $collection needed.

Comment (ctrlbold) [inline functions/Get-DBAUptime.ps1]: Agreed. FYI, I'm going to approve this one and clean it up myself to keep things going.

Comment (Stuart-Moore) [inline functions/Get-DBAUptime.ps1]: I've about half way through doing those changes if you don't want to?

Comment (ctrlbold) [inline functions/Get-DBAUptime.ps1]: Oh, okay yes please proceed. I didn't know if you were awake.

**Comment by Stuart-Moore** [comment]
New version pushed with the above revised.

**Pull request merged**